### PR TITLE
B03880 b03881 b03882 Using messaging broker to listen for updates

### PR DESCRIPTION
**Pull request by rladdusaw**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,10 +14,9 @@
   <description>An Institutional Repository IIIF manifest generation service.</description>
 
   <parent>
-    <groupId>org.springframework.boot</groupId>
-    <artifactId>spring-boot-starter-parent</artifactId>
-    <version>1.5.4.RELEASE</version>
-    <relativePath /> <!-- lookup parent from repository -->
+    <groupId>edu.tamu.weaver</groupId>
+    <artifactId>webservice-parent</artifactId>
+    <version>2.0.0</version>
```
Comment (wwelling) [inline pom.xml]: Should reference the weaver messaging module snapshot. The CI will pass when that snapshot is deployed to our maven repository.

File: src/main/java/edu/tamu/iiif/exception/UnknownActionException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package edu.tamu.iiif.exception;
```
Comment (wwelling) [inline src/main/java/edu/tamu/iiif/exception/UnknownActionException.java]: Don't need this exception.

File: src/main/java/edu/tamu/iiif/model/repo/RedisManifestRepo.java [hunk 3]
```diff
@@ -10,5 +11,9 @@
 public interface RedisManifestRepo extends CrudRepository<RedisManifest, String> {
 
     public Optional<RedisManifest> findByPathAndTypeAndRepositoryAndAllowedAndDisallowed(String path, ManifestType type, String repository, String allowed, String disallowed);
+    
+    public List<RedisManifest> findByPathAndRepository(String path, String repository);
+
+    public List<RedisManifest> findByPath(String contextPath);
```
Comment (wwelling) [inline src/main/java/edu/tamu/iiif/model/repo/RedisManifestRepo.java]: Remove unused repo named method signature.

File: src/main/java/edu/tamu/iiif/service/MessageListenerService.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    @Autowired
+    private RedisManifestRepo manifestRepo;
+
+    @Autowired
+    private List<ManifestService> manifestServices;
+
+    @WeaverMessageListener(destination = "${iiif.messaging.channel}", containerFactory = "topicContainerFactory")
+    private void update(Map<String, String> message) throws UnknownActionException, InvalidUrlException {
+        switch (message.get("action")) {
+        case "METADATA_CREATE":
```
Comment (wwelling) [inline src/main/java/edu/tamu/iiif/service/MessageListenerService.java]: We don't not want to use string literals here. Should use a enum from the weaver messaging module.

File: src/main/java/edu/tamu/iiif/service/MessageListenerService.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        case "METADATA_UPDATE":
+        case "METADATA_DELETE":
+            updateManifest(message.get("contextPath"), message.get("repositoryType"));
+            break;
+        case "RESOURCE_CREATE":
+        case "RESOURCE_DELETE":
+            updateManifest(message.get("parentContextPath"), message.get("repositoryType"));
+            break;
+        default:
+            throw new UnknownActionException("The action " + message.get("action") + "is not defined for this service");
```
Comment (wwelling) [inline src/main/java/edu/tamu/iiif/service/MessageListenerService.java]: There is nothing that is going to handle this exception. It is not needed. A log will suffice.

Messages that come through this channel that should be ignored by this service is not an exception. The log is needed to determine if the channel subscription is be saturated with unnecessary messages. If that is the case we need to be more granular about the topics and their subscription/publication.

File: pom.xml (hunk 1 above)
Comment (rladdusaw) [reply pom.xml]: https://github.com/TAMULib/IRIIIFService/commit/597c61eeeced7c98b4fe7fb925840de2c64df1b5

File: src/main/java/edu/tamu/iiif/exception/UnknownActionException.java (hunk 2 above)
Comment (rladdusaw) [reply src/main/java/edu/tamu/iiif/exception/UnknownActionException.java]: https://github.com/TAMULib/IRIIIFService/commit/b48dea90a4271025c9772504f717734f25893795

File: src/main/java/edu/tamu/iiif/model/repo/RedisManifestRepo.java (hunk 3 above)
Comment (rladdusaw) [reply src/main/java/edu/tamu/iiif/model/repo/RedisManifestRepo.java]: https://github.com/TAMULib/IRIIIFService/commit/136c070c5f2bab08042ec2bdfb6328786a3a1257

File: src/main/java/edu/tamu/iiif/service/MessageListenerService.java (hunk 4 above)
Comment (rladdusaw) [reply src/main/java/edu/tamu/iiif/service/MessageListenerService.java]: https://github.com/TAMULib/IRIIIFService/commit/136c070c5f2bab08042ec2bdfb6328786a3a1257

File: src/main/java/edu/tamu/iiif/service/MessageListenerService.java (hunk 5 above)
Comment (rladdusaw) [reply src/main/java/edu/tamu/iiif/service/MessageListenerService.java]: https://github.com/TAMULib/IRIIIFService/commit/b48dea90a4271025c9772504f717734f25893795

**Pull request merged**